src/components/Firestore/Collection.tsx

```diff
@@ -34,6 +38,13 @@ export const Collection: React.FC<Props> = ({ collection }) => {
   const [collectionSnapshot, loading, error] = useCollection(collection);
   const { url } = useRouteMatch()!;
 
+  const [isAddDocumentDialogOpen, setAddDocumentDialogOpen] = useState(false);
```

Comment:
From what I understand hooks always need to be at the top of the FC, if that's the case I'd actually consider moving this up one line and putting a line break between this hook and addDocument; mostly in the future case that someone accidentally adds something before this hook.

Reply:
Luckily eslint will warn you if you do it wrong, but it's a good suggestion, I'll do it.